### PR TITLE
[TACHYON-504] Add a simple remote read test case for Perf

### PR DESCRIPTION
**Pull request by gpang**
https://tachyon.atlassian.net/browse/TACHYON-504

A new test case for reading remote data, to test the remote read performance. Each perf task simply reads the data in a folder that another task wrote, to force remote reads.

### CODE REVIEW

**Comment by gpang** [comment]
@RongGu

File: perf/src/main/java/tachyon/perf/benchmark/simplereadremote/SimpleReadRemoteTask.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import tachyon.perf.fs.PerfFS;
+
+public class SimpleReadRemoteTask extends SimpleTask {
+  @Override
+  protected boolean setupTask(PerfTaskContext taskContext) {
+    try {
+      PerfFS fs = PerfConstants.getFileSystem();
+      // Read from another tasks directory. This will always be remote if each slave only runs one
+      // task.
+      String readDir = PerfConf.get().WORK_DIR + "/simple-read-write/" + ((mId + 1) % mTotalTasks);
```
Comment (calvinjia) [inline perf/src/main/java/tachyon/perf/benchmark/simplereadremote/SimpleReadRemoteTask.java]: Do you think total task count will be useful in the future? It seems like you could implement this by just checking if `mId` exists and if `mId + 1` doesn't exist just go to 0?

Comment (gpang) [inline perf/src/main/java/tachyon/perf/benchmark/simplereadremote/SimpleReadRemoteTask.java]: Do you think that is cleaner? I thought making the total task count explicit with the task id was ok.

**Comment by swatDong** [comment]
Do you have any idea to ensure that the read is remote when each slave runs more than one task?
This can make this test case more general.

**Comment by gpang** [comment]
Is there a simple way to ensure the read is remote?

One way to do it is to write the node name during the write test, and the read test could scan to find a directory with a different node name. I thought that was getting too complicated for a simple test, but it would be more general.

**Comment by swatDong** [comment]
How about change `mId + 1` to `mId + offset` and make the `offset` configurable?

**Comment by gpang** [comment]
I updated the PR, adding an offset config for SimpleRead. I also got rid of SimpleReadRemote, since it was basically the same as SimpleRead. Thanks @swatDong & @calvinjia !

File: perf/src/main/java/tachyon/perf/basic/PerfTask.java [hunk 2]
```diff
@@ -33,14 +33,17 @@
...
+  protected int mTotalTasks;
   protected String mNodeName;
   protected TaskConfiguration mTaskConf;
   protected String mTestCase;
 
   private PerfThread[] mThreads;
 
-  public void initialSet(int id, String nodeName, TaskConfiguration taskConf, String testCase) {
+  public void initialSet(int id, int totalTasks,
+                         String nodeName, TaskConfiguration taskConf, String testCase) {
```
Comment (calvinjia) [inline perf/src/main/java/tachyon/perf/basic/PerfTask.java]: Is the formatting here normal?

**Comment by calvinjia** [comment]
Overall LGTM, minor formatting issue?

File: perf/src/main/java/tachyon/perf/basic/PerfTask.java (hunk 2 above)
Comment (gpang) [inline perf/src/main/java/tachyon/perf/basic/PerfTask.java]: I fixed it. Thanks for catching it!

File: perf/src/main/java/tachyon/perf/benchmark/simpleread/SimpleReadTask.java [hunk 3]
```diff
@@ -28,7 +28,10 @@
   protected boolean setupTask(PerfTaskContext taskContext) {
     try {
       PerfFS fs = PerfConstants.getFileSystem();
-      String readDir = PerfConf.get().WORK_DIR + "/simple-read-write/" + mId;
+      // Potentially read from another task's directory.
+      int offset = mTaskConf.getIntProperty("taskid.offset");
+      String readDir = PerfConf.get().WORK_DIR + "/simple-read-write/"
```
Comment (apc999) [inline perf/src/main/java/tachyon/perf/benchmark/simpleread/SimpleReadTask.java]: Better to use `CommonUtils.concatPath` for portability

**Comment by apc999** [comment]
+1 LGTM

**Comment by calvinjia** [comment]
+1 LGTM

**Comment by swatDong** [comment]
+1 LGTM

**Pull request merged**